modules/alookup/alookup_test.go

```diff
@@ -263,6 +263,94 @@ func TestDoLookup(t *testing.T) {
 	} else if res != nil {
 		t.Errorf("Expected no results, got %v", res)
 	}
+
+	// Case 11: Request for A giving AAAA should return empty response
+	mockResults["example.com"] = miekg.Result{
+		Answers: []interface{}{miekg.Answer{
+			Ttl:    3600,
+			Type:   "A",
+			Class:  "IN",
+			Name:   "example.com",
+			Answer: "2001:db8::4",
+		}},
+		Additional:  nil,
+		Authorities: nil,
+		Protocol:    "",
+		Flags:       miekg.DNSFlags{},
+	}
+
+	res, _, status, _ = l.DoLookup("example.com", "")
+	if status != zdns.STATUS_NO_ANSWER {
+		t.Errorf("Expected NO_ANSWER status, got %v", status)
+	} else if res != nil {
+		t.Errorf("Expected no results, got %v", res)
+	}
+
+	// Case 12: Loops in CNAME should terminate with no results
```

Comment:
I wonder if this should terminate with no results or throw an error? For example, if we want to find how many CNAMEs chains have loops in them? Then this will hide those errors from us no? @spencerdrak any thoughts?

[Relevant RFC 1034](https://www.rfc-editor.org/rfc/rfc1034.html#section-3.6.2)
> CNAME chains should be followed and CNAME loops signalled as an error

Reply:
I would recommend checking how Google and Cloudflare deal with the loops and see if we are consistent with them?

Reply:
Returning `STATUS_ERROR` in case of loops now.